internal/component/database_observability/postgres/collector/query_samples.go

```diff
@@ -269,6 +298,29 @@ func (c *QuerySamples) fetchQuerySample(ctx context.Context) error {
 			continue
 		}
 
+		if !isIdleState(sample.State.String) {
+			c.upsertActiveSample(key, sample)
+			activeKeys[key] = struct{}{}
+			continue
+		}
+
+		// Handle idle states specially: emit finalized sample once
+		if isIdleState(sample.State.String) {
+			if st, hadActive := c.samples[key]; hadActive {
+				st.markEndedAt(sample)
+				st.LastRow.State = sample.State
+				c.emitted.Add(key, struct{}{}) // is actually emitted at the end of the loop
```

Comment:
Why can't we just set `c.samples[key]` to indicate it has been emitted? We could have some process to regularly remove

Reply:
We could, but I believe this would increase code complexity on the form of additional state checks, as the `samples` map is designed to keep in-fligh not yet finalized states only.